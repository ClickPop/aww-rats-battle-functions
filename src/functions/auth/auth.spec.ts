import request from 'supertest';
import { app } from '../..';

jest.mock('src/utils/decodeSignedMessage');

app.get('/clear-cookies', async (_, res) => {
  res.clearCookie('wallet');
  return res.send();
});

const agent = request.agent(app);

describe('/auth', () => {
  describe('/login', () => {
    test('should handle logging in', async () => {
      const res = await agent
        .post('/auth/login')
        .send({ input: { wallet: 'admin', msg: 'admin-msg' } })
        .expect(200);
      expect(res.body).toMatchSnapshot();
      expect(res.headers['set-cookie']).toBeDefined();
    });

    test('should reject logging in with an incorrect msg', async () => {
      const res = await agent
        .post('/auth/login')
        .send({ input: { wallet: 'admin', msg: 'badMsg' } })
        .expect(401);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('/', () => {
    test('should handle an incoming cookie and give back the users info', async () => {
      const res = await agent.get('/auth').expect(200);
      expect(res.body).toMatchSnapshot();
    });

    test('should return an anonymous user', async () => {
      await agent.get('/clear-cookies');
      const res = await agent.get('/auth').expect(200);
      expect(res.body).toMatchSnapshot();
    });
  });
});

describe('/auth-check', () => {
  test('should handle checking with a valid wallet', async () => {
    const res = await agent.post('/auth-check').send({
      session_variables: {
        'x-hasura-user-id': 'admin',
        'x-hasura-role': 'admin',
      },
    });
    expect(res.body).toMatchSnapshot();
  });

  test('should handle checking with anonymous', async () => {
    const res = await agent.post('/auth-check').send({
      session_variables: {
        'x-hasura-role': 'anonymous',
      },
    });
    expect(res.body).toMatchSnapshot();
  });
});
