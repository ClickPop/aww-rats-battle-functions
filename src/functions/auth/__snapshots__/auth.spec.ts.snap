// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/auth / should handle an incoming cookie and give back the users info 1`] = `
Object {
  "x-hasura-role": "admin",
  "x-hasura-user-id": "admin",
}
`;

exports[`/auth / should return an anonymous user 1`] = `
Object {
  "x-hasura-role": "anonymous",
}
`;

exports[`/auth /login should handle logging in 1`] = `
Object {
  "created_at": "2069-04-20T04:20:00+0420",
  "energy": 420,
  "id": "admin",
  "level": 69,
  "max_energy": 666,
  "updated_at": "2069-04-20T04:20:00+0420",
  "xp": 9000,
}
`;

exports[`/auth /login should reject logging in with an incorrect msg 1`] = `
Object {
  "error": "Invalid login",
}
`;

exports[`/auth-check should handle checking with a valid wallet 1`] = `
Object {
  "id": "admin",
  "role": "admin",
}
`;

exports[`/auth-check should handle checking with anonymous 1`] = `
Object {
  "id": null,
  "role": "anonymous",
}
`;
